openedx/core/djangoapps/discussions/docs/decisions/0004-transition-legacy-experience.rst

```diff
@@ -0,0 +1,56 @@
+Transition away from legacy discussions experience
+==================================================
+
+
+Status
+------
+
+Proposal
+
+
+Context
+-------
+
+To date, all courses in the LMS are automatically provided with
+Discussions via the legacy EdX system (`cs_comments_service`).
+The Discussions Tab cannot be hidden via Studio (though possibly via
+manual XML edits during Course Import?).
+
+
+Requirements
+------------
+
+- The existing discussions experience should be configurable via the new
+  Course Authoring MFE _before_ we switch to the new Discussions MFE
+  experience.
+- Existing courses should not be prompted to select a discussion
+  provider; they get the legacy experience, unless manually overridden
+  via Djano Admin.
+- Only new courses, after broad rollout, should be prompted to select an
+  available provider.
+
+
+Decision
+--------
+
+We propose to handle the transition as follows:
+
+Data
+~~~~
+- Backfill `DiscussionsConfiguration` table for existing courses
+  - If no config exists yet, create entry with `provider=legacy`
+  - TODO: What about `external_forum_url` setting?
+- Until we start allowing (requiring?) instructors to choose provider
+  - Add to `DiscussionsConfiguration` table for new courses
+- Once we require choice
```

Comment:
We still plan to default to the edx discussions as the provider, course teams are just switching this from the default.  NEw courses BEFORE the new MFE is available will default to legacy discussions. New courses AFTER our new MFE option is life will default to new discussions